// interweave/lib/shapes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MatcherShape = exports.FilterShape = undefined;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FilterShape = exports.FilterShape = _propTypes2.default.shape({
  attribute: _propTypes2.default.string.isRequired,
  filter: _propTypes2.default.func.isRequired
}); /**
     * @copyright   2016, Miles Johnson
     * @license     https://opensource.org/licenses/MIT
     * 
     */

var MatcherShape = exports.MatcherShape = _propTypes2.default.shape({
  inverseName: _propTypes2.default.string.isRequired,
  propName: _propTypes2.default.string.isRequired,
  asTag: _propTypes2.default.func.isRequired,
  createElement: _propTypes2.default.func.isRequired,
  match: _propTypes2.default.func.isRequired,
  onBeforeParse: _propTypes2.default.func,
  onAfterParse: _propTypes2.default.func
});
